packages/block-editor/src/components/default-block-appender/index.js

```diff
@@ -33,34 +34,24 @@ export function DefaultBlockAppender( {
 	const value =
 		decodeEntities( placeholder ) || __( 'Type / to choose a block' );
 
-	// The appender "button" is in-fact a text field so as to support
-	// transitions by WritingFlow occurring by arrow key press. WritingFlow
-	// only supports tab transitions into text fields and to the block focus
-	// boundary.
-	//
-	// See: https://github.com/WordPress/gutenberg/issues/4829#issuecomment-374213658
-	//
-	// If it were ever to be made to be a proper `button` element, it is
-	// important to note that `onFocus` alone would not be sufficient to
-	// capture click events, notably in Firefox.
```

Comment:
We don't need to make it a proper `button` element, `p role="button"` is fine and it doesn't suffer from the bug in Firefox.